core/arch/arm/kernel/thread.c

```diff
@@ -121,12 +122,14 @@ const uint32_t stack_tmp_stride __section(".identity_map.stack_tmp_stride") =
 DECLARE_KEEP_PAGER(stack_tmp_export);
 DECLARE_KEEP_PAGER(stack_tmp_stride);
 
-thread_pm_handler_t thread_cpu_on_handler_ptr __nex_bss;
-thread_pm_handler_t thread_cpu_off_handler_ptr __nex_bss;
-thread_pm_handler_t thread_cpu_suspend_handler_ptr __nex_bss;
-thread_pm_handler_t thread_cpu_resume_handler_ptr __nex_bss;
-thread_pm_handler_t thread_system_off_handler_ptr __nex_bss;
-thread_pm_handler_t thread_system_reset_handler_ptr __nex_bss;
+#if defined(CFG_WITH_ARM_TRUSTED_FW)
+thread_pm_handler_t thread_cpu_on_handler_ptr __nex_data = cpu_on_handler;
+thread_pm_handler_t thread_cpu_off_handler_ptr __nex_data = pm_do_nothing;
+thread_pm_handler_t thread_cpu_suspend_handler_ptr __nex_data = pm_do_nothing;
```

Comment:
Hello @jenswi-linaro , 
i do no see a way in your patch to change the function in the platform specific code anymore. Am i correct?
in the downstream optee for i.MX we are using suspend and resume handlers (currently for TZASC and CAAM configuration)
how would you suggest we should proceed?
thanks

Reply:
That's good feedback. I'll add something to this PR.
By the way, we don't have to take this PR at all. But it would be nice to get rid of all that duplicated platform specific code.

Reply:
Hello @jenswi-linaro , 
sorry for my late reply. Yes your approach (using weak functions) should be fine for the i.MX downstream use case.
thank you for the update.